typescript/vrotsc/src/compiler/transformer/codeTransformers/shims.ts

```diff
@@ -62,150 +76,153 @@ interface ShimReference {
-					return shimInstanceCall("stringIncludes", node);
-				case "String.repeat":
-					return shimInstanceCall("stringRepeat", node);
-				case "String.padStart":
-					return shimInstanceCall("stringPadStart", node);
-				case "String.padEnd":
-					return shimInstanceCall("stringPadEnd", node);
-				case "Array.find":
-					return shimInstanceCall("arrayFind", node);
-				case "Array.findIndex":
-					return shimInstanceCall("arrayFindIndex", node);
-				case "Array.fill":
-					return shimInstanceCall("arrayFill", node);
-				case "ArrayConstructor.from":
-					return shimStaticCall("arrayFrom", node);
-				case "ArrayConstructor.of":
-					return shimStaticCall("arrayOf", node);
-				case "ObjectConstructor.assign":
-					return shimStaticCall("objectAssign", node);
-			}
-		}
+    /**
+    * Visits the entire source file and replaces some built-in methods with the VROES shims.
+    *
+    * Example:
+    * ```ts
+    * "foo".startsWith("f"); // Transformed to VROES.Shims.stringStartsWith("foo", "f");
+    * "foo".endsWith("o"); // Transformed to VROES.Shims.stringEndsWith("foo", "o");
+    * "foo".includes("o"); // Transformed to VROES.Shims.stringIncludes("foo", "o");
+    * // etc...
+    * ```
+    *
+    * @param node - The node to visit.
+    * @returns - The result of visiting the node.
+    */
+    function visitCallExpression(node: ts.CallExpression): ts.Expression {
+        const symbol = context.typeChecker.getSymbolAtLocation(node.expression);
+        if (symbol) {
+            switch (context.typeChecker.getFullyQualifiedName(symbol)) {
+                case "String.startsWith":
+                    return shimInstanceCall("stringStartsWith", node);
+                case "String.endsWith":
+                    return shimInstanceCall("stringEndsWith", node);
+                case "String.includes":
+                    return shimInstanceCall("stringIncludes", node);
+                case "String.repeat":
+                    return shimInstanceCall("stringRepeat", node);
+                case "String.padStart":
+                    return shimInstanceCall("stringPadStart", node);
+                case "String.padEnd":
+                    return shimInstanceCall("stringPadEnd", node);
+                case "Array.find":
+                    return shimInstanceCall("arrayFind", node);
+                case "Array.findIndex":
+                    return shimInstanceCall("arrayFindIndex", node);
+                case "Array.fill":
+                    return shimInstanceCall("arrayFill", node);
+                case "ArrayConstructor.from":
+                    return shimStaticCall("arrayFrom", node);
+                case "ArrayConstructor.of":
+                    return shimStaticCall("arrayOf", node);
+                case "ObjectConstructor.assign":
+                    return shimStaticCall("objectAssign", node);
+                case "ObjectConstructor.setPrototypeOf":
+                    console.error("TEST!!!");
```

Comment:
Remove

Reply:
Yeap I saw that with a bit of delay hahah